score.go

```diff
@@ -99,12 +102,24 @@ func (d *indexData) scoreFile(fileMatch *FileMatch, doc uint32, mt matchTree, kn
 		}
 	}
 
+	// Add tiebreakers
+	//
+	// We multiply the score with an offset to have 5 digits for the repo rank and 2
+	// digits (plus decimals) for the doc order to function as tiebreakers.
+	fileMatch.Score = math.Trunc(fileMatch.Score) * ScoreOffset
+
 	md := d.repoMetaData[d.repos[doc]]
+	// digits 3-7 are the repo rank, this gives us increments of 1 per month.
+	addScore("repo-rank", scoreRepoRankFactor*float64(md.Rank))
+	// digits 1-2 and the decimals are the doc order.
```

Comment:
> doc-order lies in the range [0, 1), so multiplying it by 10 gives [0, 10). So wouldn't it just be "digit 1 and the decimals" ?

It is [0, 1] not [0,1). Matches in the first document of a shard will get the full score.

Reply:
> It's not obvious what range repo-rank lies in. Especially since it can represent both "months since 1970" and "number of stars (with normalization)"

I see, I will update the docs.

Reply:
> It is [0, 1] not [0,1). Matches in the first document of a shard will get the full score.

Ah, I missed the fact we do "1 - doc_id/num_docs". Thanks for clarifying the docs.